components/shared/searchInput/SearchInput.tsx

```diff
@@ -1,35 +1,72 @@
-import * as React from "react"
+"use client"
+
+import { useRouter, useSearchParams } from "next/navigation"
+import React from "react"
+import { useEffect, useRef } from "react"
 
 import { cn } from "@/lib/utils"
 
 import Icon from "../icon/Icon"
 
-// eslint-disable-next-line @typescript-eslint/no-empty-object-type
-export interface InputProps extends React.InputHTMLAttributes<HTMLInputElement> {}
-
-const SearchInput = React.forwardRef<HTMLInputElement, InputProps>(
-  ({ className, type, ...props }, ref) => {
-    return (
-      <div className="relative w-full">
-        <input
-          type={type}
-          className={cn(
-            `text-sm file:text-sm placeholder:muted-foreground flex h-20 w-full rounded-base
-            bg-background-overlay px-5 text-typo-subtitle-lg shadow-sm transition-colors file:border-0
-            file:bg-transparent file:font-medium file:text-foreground focus-visible:outline-none
-            focus-visible:ring-1 focus-visible:ring-ring disabled:cursor-not-allowed disabled:opacity-50`,
-            className
-          )}
-          ref={ref}
-          {...props}
-        />
-        <div className="absolute right-[24px] top-[50%] translate-y-[-50%]">
-          <Icon className="h-[32px] w-[32px]" name="search" />
-        </div>
-      </div>
-    )
+type SearchInputProps = {
+  className?: string
+  placeholder: string
+}
+
+const SearchInput = ({ className, placeholder }: SearchInputProps) => {
+  const inputRef = useRef<HTMLInputElement>(null)
+  const router = useRouter()
+  const searchParams = useSearchParams()
+  const [queryString, setQueryString] = React.useState("")
+
+  useEffect(() => {
+    setQueryString(searchParams.get("q") || "")
+    window.addEventListener("keydown", handleKeydown)
+    return () => {
+      window.removeEventListener("keydown", handleKeydown)
+    }
+  }, [])
+
+  const handleKeydown = (event: KeyboardEvent) => {
+    if (!inputRef.current) return
+    const focusedElement = document.activeElement as HTMLElement
+
+    if (event.key === "Enter" && focusedElement === inputRef.current) {
+      navigateToSearch()
+    }
+  }
+
+  const navigateToSearch = () => {
+    if (!inputRef.current) return
+    const inputValue = inputRef.current.value
+    router.push(inputValue ? `/search?q=${inputValue}` : "/search", {
+      scroll: false,
+    })
   }
```

Comment:
Is this accessible?
Have you tested the interface with a static A11y test or with a screen reader?
We need to be sharp on this 😅

Reply:
Great point!!! :)) but yes, it is indeed accessible by keyboard. I also added aria labels around the app now that you mentioned accessibility :)